packages/firestore/src/model/mutation.ts

```diff
@@ -484,17 +489,19 @@ export class TransformMutation extends Mutation {
     // patch, so we use the precondition to prevent incorrectly putting a
     // partial document into our cache.
     if (!this.precondition.isValidFor(maybeDoc)) {
-      return maybeDoc;
+      return new UnknownDocument(this.key, mutationResult.version);
```

Comment:
Same comments as above (axe the TODO and describe why this document must actually exist).